azext_iot/iothub/providers/message_route.py

```diff
@@ -0,0 +1,184 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import Optional
+from knack.log import get_logger
+from azure.cli.core.azclierror import ResourceNotFoundError
+from azext_iot.common.utility import process_json_arg
+from azext_iot.iothub.common import RouteSourceType
+from azext_iot.iothub.providers.base import IoTHubResourceProvider
+
+
+logger = get_logger(__name__)
+
+
+class MessageRoute(IoTHubResourceProvider):
+    def __init__(
+        self,
+        cmd,
+        hub_name: str,
+        rg: Optional[str] = None,
+    ):
+        super(MessageRoute, self).__init__(cmd, hub_name, rg)
+
+    def create(
+        self,
+        route_name: str,
+        source_type: str,
+        endpoint_name: str,
+        enabled: bool = True,
+        condition: Optional[str] = None,
+    ):
+        self.hub_resource.properties.routing.routes.append(
+            {
+                "source": source_type,
+                "name": route_name,
+                "endpointNames": endpoint_name.split(),
+                "condition": ('true' if condition is None else condition),
+                "isEnabled": enabled
+            }
+        )
+
+        return self.client.iot_hub_resource.begin_create_or_update(
+            resource_group_name=self.hub_resource.additional_properties['resourcegroup'],
+            resource_name=self.hub_resource.name,
+            iot_hub_description=self.hub_resource,
+            if_match=self.hub_resource.etag
+        )
+
+    def update(
+        self,
+        route_name: str,
+        source_type: Optional[str] = None,
+        endpoint_name: Optional[str] = None,
+        enabled: Optional[bool] = None,
+        condition: Optional[str] = None,
+    ):
+        route = self.show(route_name=route_name)
+        route.source = route.source if source_type is None else source_type
+        route.endpoint_names = route.endpoint_names if endpoint_name is None else endpoint_name.split()
+        route.condition = route.condition if condition is None else condition
+        route.is_enabled = route.is_enabled if enabled is None else enabled
+
+        return self.client.iot_hub_resource.begin_create_or_update(
+            resource_group_name=self.hub_resource.additional_properties['resourcegroup'],
```

Comment:
need consistency in ' vs "

Reply:
our preferred `black` formatter can help with that - although be warned it will format an entire document as it doesn't support "format selection"